skafka/src/main/scala/com/evolutiongaming/skafka/consumer/ConsumerOf.scala

```diff
@@ -31,20 +31,31 @@ object ConsumerOf {
       def apply[K, V](config: ConsumerConfig)(implicit fromBytesK: FromBytes[F, K], fromBytesV: FromBytes[F, V]) = {
         Consumer
           .of[F, K, V](config)
-          .map { consumer =>
-            metrics.fold { consumer } { consumer.withMetrics1[Throwable] }
+          .flatMap { consumer =>
+            metrics match {
+
+              case None =>
+                Resource.pure[F, Consumer[F, K, V]](consumer)
+
+              case Some(metrics) =>
+                for {
+                  _ <- metrics.exposeJavaMetrics[K, V](consumer)
```

Comment:
i assume this would be called on each `ConsumerOf.apply` call, right? wouldn't it try and fail to register metrics with the same name if called twice for the same consumer config?

Reply:
We have `Sync[F].delay { prometheus.unregister(collector) }` to safeguard against that, don't we? Unless it is OK to have concurrent consumers 🤔

Reply:
Yes it would, @Z1kkurat. Alternatively I can use function `ClientId => String` instead of value passed into `ConsumerMetricsOf.withJavaClientMetrics`:

```
val prefix: ClientId => String = ???
val clientId: Option[ClientId] = ???

val collector = new KafkaMetricsCollector[F](consumer.clientMetrics, clientId.map(prefix))
``` 

Client ID expected to be unique per consumer/producer, but its not enforced AFAIK. I will add explicit docs that explains connection between possible prometheus failure and client ID

Reply:
done, please review the PR again 🙏